src/scripts/main.js

```diff
@@ -1,3 +1,34 @@
 'use strict';
 
 // write code here
+
+const thead = document.querySelectorAll('thead tr th');
+const tbody = document.querySelector('tbody');
+const tbodyArr = Array.from(tbody.querySelectorAll('tr'));
+
+thead.forEach((item) => {
+  item.addEventListener('click', (e) => {
+    const columnIndex = [...thead].indexOf(e.target);
+
+    tbodyArr.sort((a, b) => {
+      const valueA = a.cells[columnIndex].textContent.trim();
+      const valueB = b.cells[columnIndex].textContent.trim();
+
+      // Перевіряємо, чи це числа
+      const numA = parseFloat(valueA.replace(/[^0-9.-]+/g, '')) || valueA;
+      const numB = parseFloat(valueB.replace(/[^0-9.-]+/g, '')) || valueB;
+
+      // Порівняння чисел або тексту
+      if (typeof numA === 'number' && typeof numB === 'number') {
```

Comment:
The type checking here is not accurate. `parseFloat` will always return a number or NaN, so the check `typeof numA === 'number'` will always be true. Instead, you should check if `!isNaN(numA)` and `!isNaN(numB)` to determine if the values are numbers.